### PR TITLE
misc refactoring

### PR DESCRIPTION
**Pull request by adamnemecek**

### CODE REVIEW

**Comment by adamnemecek** [comment]
tests succeed

**Comment by adamnemecek** [comment]
tests succeed

File: Sources/UInt128.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ public struct UInt128 {
     }
 
     public init() {
-        self.init(upperBits: 0, lowerBits: 0)
+        self = 0
```
Comment (Jitsusama) [inline Sources/UInt128.swift]: I prefer calling the initializer in this circumstance. A 0 value is quite the common operation, so I would prefer not having to go through literal translations for it.

File: Sources/UInt128.swift [hunk 2]
```diff
@@ -430,7 +431,7 @@ extension UInt128 : BinaryInteger {
 
     public init?<T : BinaryFloatingPoint>(exactly source: T) {
         if source.isZero {
-            self = UInt128()
+            self = 0
```
Comment (Jitsusama) [inline Sources/UInt128.swift]: I would prefer to call the initializer in this circumstance.

**Comment by Jitsusama** [comment]
I have a question; I know that I used to require having the underscores in order for the parameter names to not have to be specified when using those functions. Is this no longer required in Swift 4?

File: Sources/UInt128.swift [hunk 3]
```diff
@@ -398,26 +398,7 @@ extension UInt128 : BinaryInteger {
...
-            if shiftAmount > 0 {
-                shifted &>>= UInt128(upperBits: 0, lowerBits: shiftAmount)
-            }
-
-            let masked: UInt128 = shifted & UInt128(upperBits: 0, lowerBits: mask)
-
-            words.append(UInt(masked.value.lowerBits))
-        }
-        return words
+        return Array(value.lowerBits.words) + Array(value.upperBits.words)
```
Comment (adamnemecek) [inline Sources/UInt128.swift]: i wanted to put this into a separate pr but i fucked up lol. anyway, i think this is more correct than the previous implementation as for example UInt64.min.words doesn't return []

Comment (Jitsusama) [reply Sources/UInt128.swift]: Yeah, that change looks slick. Do the tests still pass, or do we need to change the tests to handle the min case?

Also, have you looked at my other comments?

**Comment by Jitsusama** [comment]
If you wouldn’t mind running tests one more time, I think this is looking good.

File: Sources/UInt128.swift (hunk 3 above)
Comment (adamnemecek) [reply Sources/UInt128.swift]: yeah, tests pass. 

i have yeah, sorry i added the other fixes after this. i still dont have a good workflow for like pushing a pr and then going back and redoing things if i have already committed other stuff.

**Comment by adamnemecek** [comment]
tests still pass.

**Comment by adamnemecek** [comment]
and re underscores, that requirement was never there for operators. like how would you even specify the argument names for like +.

File: Sources/UInt128.swift (hunk 3 above)
Comment (Jitsusama) [reply Sources/UInt128.swift]: The easiest way is to create separate branches for each PR. Rebase branches against approved PRs so that you can keep them in a consistent state, at least when the branches all deal with the same file being modified.

**Comment by Jitsusama** [comment]
Good point. I must have copied that idiom from the standard library then.
